cluster/terraform/karpenter.tf

```diff
@@ -1,77 +1,141 @@
+locals {
+  karpenter_namespace = "karpenter"
+}
+
+################################################################################
+# Controller & Node IAM roles, SQS Queue, Eventbridge Rules
+################################################################################
+
+module "karpenter" {
+  source  = "terraform-aws-modules/eks/aws//modules/karpenter"
+  version = "20.37.0"
+
+  cluster_name          = module.eks.cluster_name
+  enable_v1_permissions = true
+  namespace             = local.karpenter_namespace
+
+  # Name needs to match role name passed to the EC2NodeClass
+  node_iam_role_use_name_prefix   = false
+  node_iam_role_name              = local.name
+  create_pod_identity_association = true
+
+  tags = local.tags
+}
+
+################################################################################
+# Helm charts
+################################################################################
+
+resource "helm_release" "karpenter" {
+  name                = "karpenter"
+  namespace           = local.karpenter_namespace
+  create_namespace    = true
+  repository          = "oci://public.ecr.aws/karpenter"
+  repository_username = data.aws_ecrpublic_authorization_token.token.user_name
+  repository_password = data.aws_ecrpublic_authorization_token.token.password
+  chart               = "karpenter"
+  version             = "1.5.0"
+  wait                = false
+
+  values = [
+    <<-EOT
+    nodeSelector:
+      karpenter.sh/controller: 'true'
+    settings:
+      clusterName: ${module.eks.cluster_name}
+      clusterEndpoint: ${module.eks.cluster_endpoint}
+      interruptionQueue: ${module.karpenter.queue_name}
+    tolerations:
+      - key: CriticalAddonsOnly
+        operator: Exists
+      - key: karpenter.sh/controller
+        operator: Exists
+        effect: NoSchedule
+    webhook:
+      enabled: false
+    EOT
+  ]
+
+  lifecycle {
+    ignore_changes = [
+      repository_password
+    ]
+  }
+}
+
 # Karpenter default EC2NodeClass and NodePool
 
 resource "kubectl_manifest" "karpenter_default_ec2_node_class" {
-  yaml_body = <<YAML
-apiVersion: karpenter.k8s.aws/v1
-kind: EC2NodeClass
-metadata:
-  name: default
-spec:
-  role: "${local.node_iam_role_name}"
-  amiSelectorTerms: 
-  - alias: al2@latest
```

Comment:
might want to double check that any AL2 usage is removed in the other docs/guides prescribed in this repo